### PR TITLE
assist

### PR DESCRIPTION
**Pull request by x37v**
This PR provides both an assistance method for a wrapped class as well as new builder methods to provide assistance strings when you create iolets.

### CODE REVIEW

File: median/examples/simp/src/lib.rs [hunk 1]
```diff
@@ -33,19 +33,21 @@ median::external! {
         //create an instance of your object
         //setup inlets/outlets and clocks
         fn new(builder: &mut dyn MaxWrappedBuilder<Self>) -> Self {
+            builder.with_default_inlet_assist(&"first inlet");
```
Comment (isabelgk) [inline median/examples/simp/src/lib.rs]: Running `cargo clippy` suggests that this is a needless borrow and you can just use `builder.with_default_inlet_assist("first inlet")`

Comment (isabelgk) [reply median/examples/simp/src/lib.rs]: Also on line 41, 44, and 50

File: median/examples/hello_dsp/src/lib.rs [hunk 2]
```diff
@@ -23,8 +23,9 @@ median::external! {
     impl MSPObjWrapped<HelloDSP> for HelloDSP {
         //create some signal i/o
         fn new(builder: &mut dyn MSPWrappedBuilder<Self>) -> Self {
-            builder.add_signal_inlets(2);
-            builder.add_signal_outlets(2);
+            builder.add_signal_inlets_with_assist(&[&"in left", &"in right"]);
```
Comment (isabelgk) [inline median/examples/hello_dsp/src/lib.rs]: Clippy points out a needless borrow - could just be `(&["in left", "in right"])`

**Comment by amiel** [comment]
This looks amazing! It will be some time before I can get a chance to try it out, but I look forward to that refactor.

Do you plan to also support `inletinfo`?

**Comment by x37v** [comment]
> This looks amazing! It will be some time before I can get a chance to try it out, but I look forward to that refactor.
> 
> Do you plan to also support `inletinfo`?

yeah, I'll follow this up with inlet info

**Comment by amiel** [comment]
Wonderful!

It will be a while before I can try this out. I'm using this for a side project and I'm swamped at work and am dealing with some family illness. However, I look forward to refactoring out my hacks to use this when the time comes.

Thank you!

**Comment by amiel** [comment]
I tried this and it is working like a charm! Can't wait for `inletinfo` support as well.

**Pull request merged**